src/DependencyInjection/Compiler/ContaoRootDirPass.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2016 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Canonicalizes the contao.root_dir path.
+ *
+ * @author Leo Feyer <https://github.com/leofeyer>
+ */
+class ContaoRootDirPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $container->setParameter('contao.root_dir', realpath($container->getParameter('contao.root_dir')));
```

Comment:
>> may normalize the character case and could change unix style paths to windows style
>
> I don't think so. This is what `PathUtil::canonicalize()` does, therefore I am not using the library here.

`PathUtil::canonicalize()` converts everything into unix-style separators `/` never to windows style and does not change the character case.

But `realpath()` does according to the [PHP documentation](http://php.net/manual/en/function.realpath.php):

> For case-insensitive filesystems realpath() may or may not normalize the character case.  
> On windows realpath() will change unix style paths to windows style.